nvme_gw_server.py

```diff
@@ -86,57 +198,93 @@ def start_spdk(self):
             raise
         return
 
-    def bdev_rbd_create(self, request, context):
-        # Create bdev from a given RBD image
+    def restore_config(self):
+        callbacks = {}
+        callbacks[self.persistent_config.BDEV_PREFIX] = self.bdev_rbd_create
+        callbacks[
+            self.persistent_config.SUBSYSTEM_PREFIX] = self.nvmf_create_subsystem
+        callbacks[
+            self.persistent_config.NAMESPACE_PREFIX] = self.nvmf_subsystem_add_ns
+        callbacks[
+            self.persistent_config.HOST_PREFIX] = self.nvmf_subsystem_add_host
+        callbacks[
+            self.persistent_config.TRANSPORT_PREFIX] = self.nvmf_create_transport
+        callbacks[
+            self.persistent_config.LISTENER_PREFIX] = self.nvmf_subsystem_add_listener
+        self.persistent_config.restore(callbacks)
+
+    def bdev_rbd_create(self, request, context=None):
+        """Creates bdev from a given RBD image."""
         self.logger.info({
-            f"Received: {request.ceph_pool_name}, {request.rbd_name}, {request.block_size}",
+            f"Received request to create bdev {request.bdev_name} from",
+            f" {request.ceph_pool_name}/{request.rbd_name}",
+            f" with block size {request.block_size}",
         })
         try:
             bdev_name = self.spdk_rpc.bdev.bdev_rbd_create(
-                self.client,
-                request.ceph_pool_name,
-                request.rbd_name,
-                request.block_size,
+                self.spdk_rpc_client,
+                name=request.bdev_name,
+                pool_name=request.ceph_pool_name,
+                rbd_name=request.rbd_name,
+                block_size=request.block_size,
             )
             self.logger.info(f"Created bdev {bdev_name}")
 
         except Exception as ex:
             self.logger.error(f"bdev create failed with: \n {ex}")
-            context.set_code(grpc.StatusCode.INTERNAL)
-            context.set_details(f"{ex}")
+            if context:
+                context.set_code(grpc.StatusCode.INTERNAL)
+                context.set_details(f"{ex}")
             return pb2.bdev_info()
 
+        if context:
+            # Update persistent configuration
+            try:
+                json_req = json_format.MessageToJson(request, preserving_proto_field_name=True)
```

Comment:
For all `MessageToJson` calls -- these lines are too long:
```suggestion
                json_req = json_format.MessageToJson(
                    request, preserving_proto_field_name=True)
```